src/cascadia/TerminalApp/TerminalPage.cpp

```diff
@@ -3025,6 +3027,20 @@ namespace winrt::TerminalApp::implementation
         term.RefreshQuickFixMenu();
     }
 
+    winrt::fire_and_forget TerminalPage::_WindowSizeChanged(const IInspectable /*sender*/, const Microsoft::Terminal::Control::WindowSizeChangedEventArgs args)
+    {
+        co_await wil::resume_foreground(Dispatcher());
+
+        // Raise if:
+        // - Not in fullscreen
+        // - Only one tab exists
+        // - Only one pane exists
+        if (!_isFullscreen && NumberOfTabs() == 1 && _GetFocusedTabImpl()->GetLeafPaneCount() == 1)
```

Comment:
🛑: If we reject this resize, we need to make sure to let conpty know. We need to do something to signal a UserResize() back to the conpty. Otherwise, conpty will go on it's merry way at the new size, with the terminal at the wrong size

Reply:
Is there any existing mechanism right now that signals back to `conpty`? Maybe I can check that to have some idea. Resizing is rejected also if the window is in quake mode (in `AppHost`). Probably that will also need to signal back.

Reply:
`ConptyConnection::Resize` is the method you're looking for. The trick is just plumbing the rejection back to the correct layer to actually call that. 

It does look like `TermControl` projects it's `Connection`, so that might not be so hard to call.